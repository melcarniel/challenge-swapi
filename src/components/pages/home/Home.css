@import url('https://fonts.googleapis.com/css?family=Montserrat');

body {
  font-family: 'Montserrat';
  background-color: #E5E5E5;
  font-style: normal;
  font-weight: normal;
}

.title {
  font-size: 72px;
  line-height: 88px;
  margin-top: 10.0rem;
  margin-bottom: 0.5rem;
  color: #2C97D1;
}

.title-challenge {
  margin-bottom: 10rem;
  font-size: 14px;
  line-height: 17px;
  letter-spacing: 0.35em;
  color: #2C97D1;
}

.btn-home {
  background: #2C97D1;
  border-radius: 10px;
  color: #FFFFFF;
  padding: 1.0rem 2.6rem;
  font-size: 18px;
  line-height: 22px;
  font-weight: 900;
  border: none;
  letter-spacing: 0.35em;
  cursor: pointer;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.text-link{
  text-decoration: inherit;
}

span {
  font-weight: 950;
  line-height: 88px;
}

@media all and (max-width: 800px) {
  .container {
    margin-left: 1rem;
    margin-right: 1rem;
  }

  .title {
    font-size: 62px;
  }

  .title-challenge {
    font-size: 12px;
    margin-bottom: 8rem;
  }
  
}

