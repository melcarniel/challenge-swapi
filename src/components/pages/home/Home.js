import React, { Fragment, useEffect } from 'react';
import { Link } from 'react-router-dom';
import './Home.css'; 

const Home = () => {
  useEffect(() => {
    document.body.style.background = "#E5E5E5";
    // eslint-disable-next-line
  }, []);

  return (
    <Fragment>
      <div className="container">
        <p className="title">Welcome to <span>iClinic</span></p>
        <p className="title-challenge">FRONTEND CHALLENGE</p>
        <Link to="/result" className="btn-home text-link">START</Link>
      </div>
    </Fragment>
  )
}

export default Home;
