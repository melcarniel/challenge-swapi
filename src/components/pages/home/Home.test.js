import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Home from './Home';

configure({adapter: new Adapter()});

describe('Testes para Home', () => {

  it('Componente', () => {
    const wrapper = shallow(<Home />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});