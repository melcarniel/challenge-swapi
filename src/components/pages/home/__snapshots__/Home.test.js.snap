// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes para Home Componente 1`] = `
<Fragment>
  <div
    className="container"
  >
    <p
      className="title"
    >
      Welcome to 
      <span>
        iClinic
      </span>
    </p>
    <p
      className="title-challenge"
    >
      FRONTEND CHALLENGE
    </p>
    <Link
      className="btn-home text-link"
      to="/result"
    >
      START
    </Link>
  </div>
</Fragment>
`;

exports[`Testes para a p√°gina home fragment 1`] = `
<Fragment>
  <div
    className="container"
  >
    <p
      className="title"
    >
      Welcome to 
      <span>
        iClinic
      </span>
    </p>
    <p
      className="title-challenge"
    >
      FRONTEND CHALLENGE
    </p>
    <Link
      className="btn-home text-link"
      to="/result"
    >
      START
    </Link>
  </div>
</Fragment>
`;
